pkg/kubelet/collector/interfaces.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+type Interface interface {
+	// Starts the collector process if necessary, e.g., the builtin cadvisor
+	// Otherwise, does nothing and just returns
+	Start() error
+
+	// Returns the basic machine spec
+	// Currently used to report to k8s master the basic information about a node
+	MachineInfo() (*MachineInfo, error)
+
+	// Returns the software version of various components of a node, e.g., kernel, container runtime, etc.
+	// Currently used to report back to k8s master the versions of relevant software on a node
+	VersionInfo() (*VersionInfo, error)
+
+	// Returns the file system information of a given label or mount point
```

Comment:
mount points are in absolute path, i.e. starting with '/', and labels are not. cadvisor currently only allows fs queries based on labels, but I think the interface should be more generically defined and allow queries based on mount points. I will add the distinction between labels and mounts points in the comments.